Refs #27753 -- Deprecated django.utils.text.unescape_entities().

jdufresne opened the pull request:
The function was undocumented and only required for compatibility with Python 2.

Use Python's html.unescape() instead -- available since Python 3.4.

https://docs.python.org/3/library/html.html#html.unescape

https://code.djangoproject.com/ticket/27753

bluetech commented on docs/releases/3.0.txt:
```diff
@@ -329,6 +329,9 @@ Django 3.0, we're removing these APIs at this time.
 
 * ``django.utils.safestring.SafeBytes`` - Unused since Django 2.0.
 
+* ``django.utils.text.unescape_entities()`` - Use :func:`html.unescape`
```
Django's `unescape_entities` had `@keep_lazy_text` applied. For Django's own use of the function in `multipartparser.py` I don't think it is needed, but maybe it's not a straight replacement for other users?

jdufresne commented:
Thanks for the review. As suggested, I've changed the PR to simply deprecate the function rather than removing it. I also added a note about lazy strings.

felixxm commented:
@jdufresne Thanks for this PR :+1: Please add also a note to the `docs/internals/deprecation.txt`.

jdufresne commented:
Done. :slightly_smiling_face:

jdufresne commented on docs/releases/3.0.txt:
```diff
@@ -411,6 +411,9 @@ Miscellaneous
   :meth:`.Expression.get_group_by_cols`.
 
+* ``django.utils.text.unescape_entities()`` is deprecated in favor of
+  :func:`html.unescape`. For lazy strings, first coerce using ``str()``.
```
Excellent point. I've used your wording in the latest revision.
